.github/workflows/rust.yml

```diff
@@ -92,3 +96,32 @@ jobs:
       with:
         header: bench
         path: pr-comment
+
+  check-publish-key-share:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: Swatinem/rust-cache@v2
+      with:
+        cache-on-failure: "true"
+    - name: Dry-run publish
+      run: cargo publish --dry-run -p key-share
+  check-publish-cggmp21-keygen:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: Swatinem/rust-cache@v2
+      with:
+        cache-on-failure: "true"
+    - name: Dry-run publish
+      run: cargo publish --dry-run -p cggmp21-keygen
+  # TODO: publish the crates above and add the check below (doesn't work without them)
+  #check-publish-cggmp21:
+  #  runs-on: ubuntu-latest
+  #  steps:
+  #  - uses: actions/checkout@v3
+  #  - uses: Swatinem/rust-cache@v2
+  #    with:
+  #      cache-on-failure: "true"
+  #  - name: Dry-run publish
+  #    run: cargo publish --dry-run -p cggmp21
```

Comment:
Use a matrix instead: see [example](https://github.com/dfns/generic-ec/blob/df7501ec4d2d686be96fc8f5bc8f75599ceab91a/.github/workflows/rust.yml#L121-L136) in generic-ec